Docs: polish README.md (#656)
devcontainers/images: src/python/README.md

### Configuration

> You can directly reference pre-built versions of this image by using the `image` property in `.devcontainer/devcontainer.json` or updating the `FROM` statement in your own `Dockerfile` with one of the following:

Reviewer: ```suggestion
You can directly reference pre-built versions of this [.devcontainer](.devcontainer) by using the `image` property in `.devcontainer/devcontainer.json` or updating the `FROM` statement in your own `Dockerfile` with one of the following:
```

Curious, does this make more sense? 
// cc @bamurtaugh

Reviewer 2: Thanks for tagging me! I could see either making sense. I think `image` makes a bit more sense to me as we don't really use `.devcontainer` as a term elsewhere.

Author: I'd definitely stick with `image` here, because that is what is meant in this sentence (IIUC), i.e. the [Docker image](https://docs.docker.com/glossary/#image), not the instructions of how to build that image, i.e. "traditionally" the `Dockerfile`, here the `Dockerfile` + `devcontainer.json`.

`pre-built` is a bit awkward maybe, because images are by definition built, so maybe 

```suggestion
You can directly reference one of these images by using the `image` property in `.devcontainer/devcontainer.json` or updating the `FROM` statement in your own `Dockerfile` with one of the following:
```

Reviewer 2: This is great feedback. I think we'd still want to say "pre-built," but maybe we could link to our current docs content on pre-builds for more info? https://containers.dev/implementors/reference/#prebuilding

We also have an issue where @craiglpeters and I are going to work on highlighting and explaining pre-builds more clearly: https://github.com/devcontainers/spec/issues/261.

Author: Yes, this makes much more sense. So we do the following?

```suggestion
You can directly reference [pre-built](https://containers.dev/implementors/reference/#prebuilding) versions of this image by using the `image` property in `.devcontainer/devcontainer.json` or updating the `FROM` statement in your own `Dockerfile` with one of the following:
```

Reviewer 2: LGTM!

- `mcr.microsoft.com/devcontainers/python:1.0-3.9` (or `1.0-3.9-bullseye`, `1.0-3.9-buster`)
- `mcr.microsoft.com/devcontainers/python:1.0.0-3.9` (or `1.0.0-3.9-bullseye`, `1.0.0-3.9-buster`)

However, we only do security patching on the latest [non-breaking, in support](https://github.com/devcontainers/images/issues/90) versions of images (e.g. `1-14`). 
You may want to run `apt-get update && apt-get upgrade` in your Dockerfile if you lock to a more specific version to at least pick up OS security updates.

See [history](history) for information on the contents of each version and [here for a complete list of available tags](https://mcr.microsoft.com/v2/devcontainers/python/tags/list).

Alternatively, you can use the contents of [.devcontainer](.devcontainer) to fully customize the your container's contents or build for a container architecture the image does not support.

> Beyond Python and `git`, this image includes a number of Python tools, `zsh`, [Oh My Zsh!](https://ohmyz.sh/), a non-root `vscode` user with `sudo` access, and a set of common dependencies for development, see [.devcontainer/devcontainer.json](.devcontainer/devcontainer.json).

Author: I added this link because I think it might be helpful to see where these additions to the image happen (if you are, like me currently, new to devcontainers).

Reviewer 2: Thanks for sharing this! I'm not sure I'd add the devcontainer.json pointer. It's definitely a key file, but not the only one that includes tools or dependencies, so it may be a bit confusing.

Author: Well, I wasn't aware that some of the build instructions (could) happen in devcontainer.json before @samruddhikhandale explained it in #653. If you're used to devcontainers this is obvious, I guess, but not necessarily if you're not IMHO. So at the risk of adding this for an audience of one (i.e. me), I'd say it might not be perfect, but still better ...? 

I am curious: Where are other tools and deps for this image defined, apart from `.devcontainer/devcontainer.json` and `.devcontainer/Dockerfile`.

Reviewer 2: Thanks for the great context here - it's always super helpful hearing where things are unclear or could be better.

What about mentioning the .devcontainer folder in general, rather than .devcontainer/devcontainer.json? That way it indicates the config is contained in essentially any files in this folder (which could be devcontainer.json, Dockerfile, Docker Compose).

Author: It looses a bit of the (my) 🤯 of "the devcontainer.json (also) defines what's in the image" but this is indeed not the right place for this. So yeah, let's do 

```suggestion
Beyond Python and `git`, this image includes a number of Python tools, `zsh`, [Oh My Zsh!](https://ohmyz.sh/), a non-root `vscode` user with `sudo` access, and a set of common dependencies for development, see [.devcontainer](.devcontainer).
```

if that's still fine with you!

Reviewer 2: Looks great, thank you! As a small nit, this might flow a little more clearly for me:

```suggestion
Beyond Python and `git`, this image includes a number of Python tools, `zsh`, [Oh My Zsh!](https://ohmyz.sh/), a non-root `vscode` user with `sudo` access, and a set of common dependencies for development; see [.devcontainer/devcontainer.json](.devcontainer/devcontainer.json).
```

or

```suggestion
Beyond Python and `git`, this image includes a number of Python tools, `zsh`, [Oh My Zsh!](https://ohmyz.sh/), a non-root `vscode` user with `sudo` access, and a set of common dependencies for development. See [.devcontainer/devcontainer.json](.devcontainer/devcontainer.json) for the full config.
```

Author: You probably mean

```suggestion
Beyond Python and `git`, this image includes a number of Python tools, `zsh`, [Oh My Zsh!](https://ohmyz.sh/), a non-root `vscode` user with `sudo` access, and a set of common dependencies for development. See [.devcontainer](.devcontainer) for the full config.
```

i.e. the link to the folder?! 😄

Reviewer 2: Haha yes, going too fast yesterday 😄

- `mcr.microsoft.com/devcontainers/python:1.0-3.9` (or `1.0-3.9-bullseye`, `1.0-3.9-buster`)
- `mcr.microsoft.com/devcontainers/python:1.0.0-3.9` (or `1.0.0-3.9-bullseye`, `1.0.0-3.9-buster`)

> However, we only do security patching on the latest [non-breaking, in support](https://github.com/devcontainers/images/issues/90) versions of images (e.g. `1-14`). 

Reviewer: ```suggestion
However, we only do security patching on the latest [non-breaking, in support](https://github.com/devcontainers/images/issues/90) versions of images (e.g. `1-3`). 
```
